### ISSUE TITLE
VSCodeAzureSubscriptionProvider.getSubscriptions return empty

### ISSUE THREAD

**Issue by jianyexi** [microsoft/vscode-azuretools#1815]
When sign in with below code , the VSCodeAzureSubscriptionProvider.isSignedIn() return true, 
```
vscode.authentication.getSession("microsoft", scopes, { createIfNone: true });
```

But the `VSCodeAzureSubscriptionProvider.getSubscriptions()`  return empty,  as below code will always return false. I think we need to improve the 
`VSCodeAzureSubscriptionProvider.isSignedIn` method to handle the current signed tenantId in this case as well
https://github.com/microsoft/vscode-azuretools/blob/f92ce31fe6ff96e8827e598700dee010bcce8ef8/auth/src/VSCodeAzureSubscriptionProvider.ts#L111

**Comment by alexweininger** [comment]
Hi @jianyexi, yes this seems like an issue we need to fix. Have you found a workaround yet or do you need me to investigate and unblock you? If you need, reach out to me on Teams.

**Comment by jianyexi** [comment]
> Hi [@jianyexi](https://github.com/jianyexi), yes this seems like an issue we need to fix. Have you found a workaround yet or do you need me to investigate and unblock you? If you need, reach out to me on Teams.

No, I can't find a workaround, I am considering to implement the function of listing subscriptions through the ARM Management SDK, but  I will appreciate if it could be fixed in short time.

**Comment by alexweininger** [comment]
@jianyexi, can you check if your issue is fixed in v3.0.1? https://github.com/microsoft/vscode-azuretools/releases/tag/microsoft-vscode-azext-azureauth-v3.0.1

**Comment by jianyexi** [comment]
@alexweininger no, the `getSessionFromVSCode` will return null  in my case.

**Comment by alexweininger** [comment]
> Retrieve the tenantId from the account.Id

I don't think this is possible, `account.id` doesn't always contain the default tenant id.

If the user needs to sign in to a specific tenant in order to list subscriptions, we're going to need you to handle that in your extension. We can work together for a better solution in the long term, but until then we need you to handle that on your own.

There are some methods in the auth package that can help you do this, for example `getUnauthenticatedTenants` lists the tenants we don't have sessions for. You can call that method if `getSubscriptions` returns empty and ask the user to sign into one of those tenants.

**Comment by alexweininger** [comment]
This issue is fixed by #1827
